flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorProvider.java

```diff
@@ -81,11 +81,13 @@ public OperatorCoordinator getCoordinator(OperatorCoordinator.Context context) t
 	 */
 	public static class CoordinatorExecutorThreadFactory implements ThreadFactory {
 		private final String coordinatorThreadName;
+		private final ClassLoader cl;
 		private Thread t;
 
 		CoordinatorExecutorThreadFactory(String coordinatorThreadName) {
 			this.coordinatorThreadName = coordinatorThreadName;
 			this.t = null;
+			this.cl = Thread.currentThread().getContextClassLoader();
```

Comment:
I think it would be safer to pass the user code classloader here directly, rather than relying on the context class loader. That seems fragile.